src/Sarif.Viewer.VisualStudio.Core/Services/DataService.cs

```diff
@@ -48,34 +50,29 @@ public void SendEnhancedResultData(SarifLog sarifLog)
             SendEnhancedResultDataAsync(sarifLog).FileAndForget(Constants.FileAndForgetFaultEventNames.SendEnhancedData);
         }
 
-        private static async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
+        private async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
         {
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
             Assumes.NotNull(sarifLog);
             Assumes.True(sarifLog.Runs?.Count == 1);
 
-            var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
-            if (componentModel != null)
-            {
-                var dte = Package.GetGlobalService(typeof(DTE)) as DTE2;
-                var projectNameCache = new ProjectNameCache(dte?.Solution);
-                var items = new List<SarifErrorListItem>();
-                Run run = sarifLog.Runs.First();
+            string logPath = $"{Guid.NewGuid()}.sarif";
```

Comment:
>logPath [](http://example.com/codeflow?start=19&length=7)

If they send us the same file again, will we have duplicate error list items?

Reply:
how do we identify each key event Sarif log?
if we know the identifier, we can close the items from previous call, and re-open all items from new call if they have same identifier.

Reply:
hi @kylereedmsft 

if the same key event log send to viewer twice, should we show 2 error list items? 

if the 2nd key event log send to viewer, with different key event data compared to previous one, should we refresh previous error list item?

with an identifier we can control what error list items we should update (remove then add again). if not, we need to remove all existing key event error list items for each key event log sent, and create new error list items from the log. 

Had a discussion with Chris he will send some questions regarding the scenario.

Reply:
When we send a log via the interop path, we do not want anything added to the error list.
Visual Studio already has the items in the error list because of background code analysis so if your code adds them too, there will be duplicate entries.

Reply:
Understood, we will make the change.

Reply:
sorry for the confusing.. updated the fix pls review